src/testing/flow_solver_cases/periodic_cube_flow.cpp

```diff
@@ -98,85 +112,22 @@ double PeriodicTurbulence<dim,nstate>::get_constant_time_step(std::shared_ptr<DG
-        unsteady_data_table->set_precision(time_string, 16);
-        unsteady_data_table->set_scientific(time_string, true);
-        // Add kinetic energy to table
-        std::string kinetic_energy_string = "kinetic_energy";
-        unsteady_data_table->add_value(kinetic_energy_string, kinetic_energy);
-        unsteady_data_table->set_precision(kinetic_energy_string, 16);
-        unsteady_data_table->set_scientific(kinetic_energy_string, true);
-        // Write to file
-        std::ofstream unsteady_data_table_file(this->unsteady_data_table_filename_with_extension);
-        unsteady_data_table->write_text(unsteady_data_table_file);
-    }
-    // Print to console
-    this->pcout << "    Iter: " << current_iteration
-                << "    Time: " << current_time
-                << "    Energy: " << kinetic_energy
-                << std::endl;
-
-    // Abort if energy is nan
-    if(std::isnan(kinetic_energy)) {
-        this->pcout << " ERROR: Kinetic energy at time " << current_time << " is nan." << std::endl;
-        this->pcout << "        Consider decreasing the time step / CFL number." << std::endl;
-        std::abort();
-    }
-}
-
-template<int dim, int nstate>
-double PeriodicTurbulence<dim,nstate>::integrand_kinetic_energy(const std::array<double,nstate> &soln_at_q) const
-{
-    // Description: Returns nondimensional kinetic energy
-    const double nondimensional_density = soln_at_q[0];
-    double dot_product_of_nondimensional_momentum = 0.0;
-    for (int d=0; d<dim; ++d) {
-        dot_product_of_nondimensional_momentum += soln_at_q[d+1]*soln_at_q[d+1];
-    }
-    const double nondimensional_kinetic_energy = 0.5*(dot_product_of_nondimensional_momentum)/nondimensional_density;
-    return nondimensional_kinetic_energy/this->domain_size;
-}
-
 template<int dim, int nstate>
-double PeriodicTurbulence<dim,nstate>::integrand_l2_error_initial_condition(const std::array<double,nstate> &soln_at_q, const dealii::Point<dim> qpoint) const
+void PeriodicTurbulence<dim, nstate>::compute_and_update_integrated_quantities(DGBase<dim, double> &dg)
 {
-    // Description: Returns l2 error with the initial condition function
-    // Purpose: For checking the initialization
-    std::shared_ptr<InitialConditionFunction<dim,nstate,double>> initial_condition_function = InitialConditionFactory<dim,nstate,double>::create_InitialConditionFunction(&this->all_param);
-    double integrand_value = 0.0;
-    for (int istate=0; istate<nstate; ++istate) {
-        const double exact_soln_at_q = initial_condition_function->value(qpoint, istate);
-        integrand_value += pow(soln_at_q[istate] - exact_soln_at_q, 2.0);
-    }
-    return integrand_value;
-}
-
-template<int dim, int nstate>
-double PeriodicTurbulence<dim, nstate>::integrate_over_domain(DGBase<dim, double> &dg,const IntegratedQuantitiesEnum integrated_quantity) const
-{
-    double integral_value = 0.0;
+    this->MAX_NUMBER_OF_COMPUTED_QUANTITIES = 4;
```

Comment:
Unused member variable. It's ok to just have a `constexpr int MAX_NUMBER_OF_COMPUTED_QUANTITITES = 4` locally since it's specific to this test.

Reply:
Or find a way to use `IntegratedQuantitiesEnum::INTEGRATEDQUANTITIESENUM_NR_ITEMS` as a compile-time constant to use in the array sizes.

Reply:
fixed